test/fuzz/transformation_add_float_type_test.cpp

```diff
@@ -0,0 +1,134 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "source/fuzz/transformation_add_type_float.h"
+
+#include "gtest/gtest.h"
+#include "source/fuzz/fuzzer_util.h"
+#include "test/fuzz/fuzz_test_util.h"
+
+namespace spvtools {
+namespace fuzz {
+namespace {
+
+TEST(TransformationAddConstantBooleanTest, NeitherPresentInitiallyAddBoth) {
+  std::string shader = R"(
+               OpCapability Shader
+          %1 = OpExtInstImport "GLSL.std.450"
+               OpMemoryModel Logical GLSL450
+               OpEntryPoint Fragment %4 "main"
+               OpExecutionMode %4 OriginUpperLeft
+               OpSource ESSL 310
+               OpName %4 "main"
+          %2 = OpTypeVoid
+          %3 = OpTypeFunction %2
+          %4 = OpFunction %2 None %3
+          %5 = OpLabel
+               OpReturn
+               OpFunctionEnd
+  )";
+
+  const auto env = SPV_ENV_UNIVERSAL_1_3;
+  const auto consumer = nullptr;
+  const auto context = BuildModule(env, consumer, shader, kFuzzAssembleOption);
+  spvtools::ValidatorOptions validator_options;
+  ASSERT_TRUE(fuzzerutil::IsValidAndWellFormed(context.get(), validator_options,
+                                               kConsoleMessageConsumer));
+  TransformationContext transformation_context(
+      MakeUnique<FactManager>(context.get()), validator_options);
+  
+  // Should be able to add a 32-bit float type with id 7
+  ASSERT_TRUE(TransformationAddTypeFloat(7, 32)
+                  .IsApplicable(context.get(), transformation_context));
+  
+
+  // Should be able to add a 16-bit float type with id 7
+  if (!context.get()->get_feature_mgr()->HasCapability(SpvCapabilityFloat16)) {
+       ASSERT_FALSE(TransformationAddTypeFloat(7, 16)
+          .IsApplicable(context.get(), transformation_context));
+  }
+  
+  else{
+
+     ASSERT_TRUE(TransformationAddTypeFloat(7, 16)
+          .IsApplicable(context.get(), transformation_context));
+     
+  }
+  
+  // Should be able to add a 64-bit float type with id 7
+  if (!context.get()->get_feature_mgr()->HasCapability(SpvCapabilityFloat64)) {
+       ASSERT_FALSE(TransformationAddTypeFloat(7, 64)
+          .IsApplicable(context.get(), transformation_context));
+  }
+  
+  else{
+
+     ASSERT_TRUE(TransformationAddTypeFloat(7, 64)
+          .IsApplicable(context.get(), transformation_context));
+     
+  }
+
+
+  // Id 2 already exists so the statement below should return false
+  ASSERT_FALSE(TransformationAddTypeFloat(2, 32)
+                  .IsApplicable(context.get(), transformation_context));
+
+  
+  // Should not be able to add float type of width different from 16/32/64
+  // This test should should fail but it doesn't because of the
+  // assert statement in the default case of TransformationAddTypeFloat::IsApplicable
+//   ASSERT_FALSE(TransformationAddTypeFloat(7, 20)
+//                   .IsApplicable(context.get(), transformation_context));
```

Comment:
Turn this into a "death test". Search for ASSERT_DEATH in other test files and you'll see how to do it.